scalafix-core/src/main/scala/scalafix/rewrite/NoValInForComprehension.scala

```diff
@@ -0,0 +1,21 @@
+package scalafix
+package rewrite
+
+import scala.meta._
+import scala.meta.contrib._
+
+case object NoValInForComprehension extends Rewrite {
+
+  override def rewrite(ctx: RewriteCtx): Patch = {
+    ctx.tree.collect {
+      case f: Term.ForYield =>
```

Comment:
I believe this rewrite should also work for `Term.For`. We can make an extractor

```scala
  object TermFor {
    def unapply(tree: Tree): Option[Seq[Enumerator]] = tree match {
      case Term.For(enums, _) => Some(enums)
      case Term.ForYield(enums, _) => Some(enums)
      case _ => None
    }
  }
```